Several additional Tests with Ajax dependencies, adding Selenium implementation!

werpu opened the pull request:
for #1722

While testing the new MyFaces codebase.,
Several people in the project noticed a set of Ajax dependent TCK tests, which still had no Selenium implementation.
This pull request adds implementation for the "discovered" tests!

Following Tests are affected:
ee.jakarta.tck.faces.test.javaee6.viewParamNullValueAjax.Issue4550IT.testViewParamNullValueAjax
ee.jakarta.tck.faces.test.javaee8.cdi.Spec1351IT.testInjectValidator
ee.jakarta.tck.faces.test.javaee8.cdi.Spec1386IT.testInjectFlowMap
ee.jakarta.tck.faces.test.javaee8.commandScript.Spec613IT.test
ee.jakarta.tck.faces.test.servlet40.exactmapping.Spec1260IT.testLinkToNonExactMappedView
ee.jakarta.tck.faces.test.servlet40.exactmapping.Spec1260IT.testAjaxFromExactMappedView
ee.jakarta.tck.faces.test.servlet40.exactmapping.Spec1260IT.testExactMappedViewLoads
ee.jakarta.tck.faces.test.servlet40.exactmapping.Spec1260IT.testPostBackOnLinkedNonExactMappedView
ee.jakarta.tck.faces.test.servlet40.exactmapping.Spec1260IT.testPostBackToExactMappedView
ee.jakarta.tck.faces.test.servlet40.exactmapping.Spec1260IT.testResourceReferenceFromExactMappedView
ee.jakarta.tck.faces.test.servlet40.facelets.Issue4830IT.testUIRepeatResetValues
ee.jakarta.tck.faces.test.servlet40.facelets.Issue5078IT.testUIRepeatVisitTreeDuringInvokeApplication
ee.jakarta.tck.faces.test.javaee8.searchExpression.Issue4331IT.test
ee.jakarta.tck.faces.test.javaee8.searchExpression.Spec1238IT.test
ee.jakarta.tck.faces.test.javaee8.uiinput.Issue5081IT.testIssue4734
ee.jakarta.tck.faces.test.servlet50.inputfile.Spec1555IT.testMultipleSelectionNonAjax
ee.jakarta.tck.faces.test.servlet50.inputfile.Spec1555IT.testSingleSelectionNonAjax
ee.jakarta.tck.faces.test.servlet50.inputfile.Spec1555IT.testMultipleSelectionAjax
ee.jakarta.tck.faces.test.servlet50.inputfile.Spec1555IT.testSingleSelectionAjax

werpu committed:
Several tests had ajax dependencies, moving them over to Selenium!

werpu committed:
Fixing missing test skips on test.selenium=true

volosied commented on tck/faces40/inputFile/src/test/java/ee/jakarta/tck/faces/test/servlet50/inputfile/Spec1555IT.java:
```diff
@@ -48,7 +49,7 @@
 import jakarta.faces.component.html.HtmlInputFile;
 
-@RunWith(Arquillian.class)
+@RunWith(BaseArquilianRunner.class)
```
@werpu  As this is tagged with 4.0.1, could you undo this change?  It should be done a separate PR.

werpu replied:
I can undo it but then the Spec1555IT will be triggered always. That means the test will fail on myfaces because it triggers HTMLUnit with test.selenium=true
If wanted I will isolate the two classes which are not fit for 4.01 and issue a separate PR tomorrow.
Anything else which needs isolation for the 4.0.1 tagging?

brideck replied:
I think Vlad may have tagged the wrong file here.  The BeanValidation test is the one that would want to be in a separate PR -- tck/faces40/beanValidation/src/test/java/ee/jakarta/tck/faces/test/servlet50/beanValidation/Issue5171IT.java

That way 4.0.1 changes can be siloed from changes to new tests.

volosied replied:
Oops sorry about that! 😞

werpu replied:
Ok I will isolate the bean validation change into a separate non 4.0.1 pr later today!
Thanks for the update!

werpu replied:
Hi I have isolated the BeanValidation fix in https://github.com/jakartaee/faces/pull/1797 the changes on the test are dropped from the this pull request!

werpu committed:
Taking out the bean validation fix, since it is tagged for 4.0.1 and will not make it in. It will be issued in a separate pull request.
Resetting the file to the master state!

werpu committed:
Relicensing of the base framework as EPL-2.0
Note, I will relicense the same codebase in another project as BSD or ASL2 as long as it is the code I have written myself, but for Jakarta it is EPL-2.0 to have this unified under one license!

werpu committed:
Eliminating the converter, it is not part of the framework but a helper class speeding up test conversion privately used!

werpu committed:
Relicensing under EPL, GPL2 and ASL2 combination license!

Adding missing EPL license header in ITBase (code which has been existing, so no ASL2 combination license possible)
